core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -260,17 +260,24 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                     final long maxContentLength = decodedReq.maxRequestLength();
                     final long transferredLength = decodedReq.transferredBytes();
                     if (maxContentLength > 0 && transferredLength > maxContentLength) {
-                        final ContentTooLargeException cause =
-                                ContentTooLargeException.builder()
-                                                        .maxContentLength(maxContentLength)
-                                                        .contentLength(req.headers())
-                                                        .transferred(transferredLength)
-                                                        .build();
-                        fail(id, decodedReq.headers(), HttpStatus.REQUEST_ENTITY_TOO_LARGE,
-                             Http2Error.CANCEL, null, cause);
-                        // Wrap the cause with the returned status to let LoggingService correctly log the
-                        // status.
-                        decodedReq.close(HttpStatusException.of(HttpStatus.REQUEST_ENTITY_TOO_LARGE, cause));
+                        final Routed<ServiceConfig> routed = req.route();
+                        if (routed != null && routed.route().isFallback()) {
```

Comment:
Question about the changed behavior:

If I understand correctly, when `decodedReq.needsAggregation()` is true then `ctx.fireChannelRead()` would be invoked late meaning that a client would need to wait until the entire request is received before receiving a 404 instead of an immediate 413.

Is this change intentional?

Reply:
`FallbackService` does not set `decodedReq.needsAggregation()` to true. Did you mean a non-request service that returns 404 for a certain path?
The behavior of returning 404 for non-existent service has not changed.
https://github.com/line/armeria/blob/dc700253ad912f099445437cd5008b77c326b7fe/core/src/test/java/com/linecorp/armeria/server/HttpServerTest.java#L598-L605

Reply:
> FallbackService does not set decodedReq.needsAggregation() to true.

Oops, I think you're right 😅 

Still, for instance if a global decorator is handling the request instead of `FallbackService` I'm wondering if just returning a 413 is more appropriate 🤔

Reply:
Sorry, I didn't get this. It meant if a decorator is in the middle of handling a request, do we need to return 413? I'm not sure we can determine a response status by detecting a decorator which is handling a non-existent URL.

Reply:
I think you can ignore this comment, I think I misunderstood how the PR works 😅 

For some reason, I thought `atomicLong` in the following snippet would overflow over `maxRequestLength`.
It turns out with the current diff, all additional `HttpContent` will be ignored, so we don't have to worry about this case.

Sorry about the confusion 😅 

```
sb.decorator((delegate, ctx, req) -> {
    final AtomicLong atomicLong = new AtomicLong();
    req.subscribe(new Subscriber<HttpObject>() {
        @Override
        public void onSubscribe(Subscription s) {
            s.request(Long.MAX_VALUE);
        }

        @Override
        public void onNext(HttpObject httpObject) {
            if (httpObject instanceof HttpData) {
                atomicLong.addAndGet(((HttpData)httpObject).length());
                logger.info("current content size: {}", atomicLong);
            }
        }
    });
    return HttpResponse.delayed(HttpResponse.of(200), Duration.ofSeconds(3));
});
```